### ISSUE TITLE
containsRange checks are not symmetric with explicit subnet masks involved

### ISSUE THREAD

**Issue by voetsjoeba** [mlocati/ip-lib#75]
An edge case for `containsRange` I ran into; I was expecting both these checks to return true:

```
use IPLib\Factory;

$a = Factory::rangeFromString("1.2.3.4");
$b = Factory::rangeFromString("1.2.3.4/32");

$a->containsRange($b); // false
$b->containsRange($a); // true
```

For context, this came up during some blacklisting logic, where I want to check a user-provided input address or range against a predefined list of excluded IPs or ranges. Both sides are 'input' in the sense that they could come from configuration or from users, so either side could conceivably contain an explicit /32 suffix.

**Comment by voetsjoeba** [comment]
Appears to be due to `IPLib\Range\Single`'s implementation of `containsRange` running an exact string comparison between its own string representation and the given range's, but its own string representation doesn't include the /32.

Workaround that appears to work is to explicitly normalize both sides to an `IPLib\Range\Subnet` instance instead, which instead performs the standard logic of comparing start and end addresses in its `containsRange` implementation:

```
use IPLib\Factory;

$a = Factory::rangeFromString("1.2.3.4");
$b = Factory::rangeFromString("1.2.3.4/32");

$a->asSubnet()->containsRange($b); // true
$b->asSubnet()->containsRange($a); // true
```

Still though, I would've expected the original logic to work without a normalization step.

**Comment by mlocati** [comment]
Thanks for taking the time to report this issue.
I've just published version 1.17.1 that fixes this issue.
